<launch>
	<!-- =================== SET Argument =================== -->
	<arg name="camera_id_right" value="2"/>
	<arg name="ext_trigger_mode" value="False" />
	<arg name="color_mode" value="bgr8" />
	<arg name="image_width" value="1936" />
	<arg name="image_height" value="1216" />
	<arg name="auto_gain" value="False"/>
	<arg name="master_gain" value="10" />
	<arg name="red_gain" value="100" />
	<arg name="green_gain" value="0" />
	<arg name="blue_gain" value="20" />
	<arg name="auto_exposure" value="False"/>
	<arg name="exposure" value="10"/>
	<arg name="auto_white_balance" value="False"/>
	<arg name="auto_frame_rate" value="False"/>
	<arg name="frame_rate" value="10.0"/>
	<arg name="pixel_clock" value="30"/>
	<arg name="flip_lr" value="False" />
	<arg name="nodelet_manager_name_right" value="stereo_right" />
	<arg name="camera_name_right" value="/stereo/right" />

	<!-- =================== RIGHT CAMERA =================== -->
	<node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name_right)" args="manager" output="screen" />
	<node pkg="nodelet" type="nodelet" name="ueye_cam_nodelet_$(arg nodelet_manager_name_right)" args="load ueye_cam/ueye_cam_nodelet $(arg nodelet_manager_name_right)">
		<param name="camera_name" type="str" value="$(arg camera_name_right)" />
		<param name="camera_topic" type="str" value="image_raw" />
		<param name="camera_id" type="int" value="$(arg camera_id_right)" />
		<param name="ext_trigger_mode" type="bool" value="$(arg ext_trigger_mode)" />
		<param name="color_mode" type="str" value="$(arg color_mode)" />
		<param name="image_width" type="int" value="$(arg image_width)" />
		<param name="image_height" type="int" value="$(arg image_height)" />
		<param name="auto_gain" type="bool" value="$(arg auto_gain)" />
		<param name="master_gain" type="int" value="$(arg master_gain)" />
		<param name="red_gain" type="int" value="$(arg red_gain)" />
		<param name="green_gain" type="int" value="$(arg green_gain)" />
		<param name="blue_gain" type="int" value="$(arg blue_gain)" />
		<param name="auto_exposure" type="bool" value="$(arg auto_exposure)" />
		<param name="exposure" type="int" value="$(arg exposure)" />
		<param name="auto_white_balance" type="bool" value="$(arg auto_white_balance)" />
		<param name="auto_frame_rate" type="bool" value="$(arg auto_frame_rate)" />
		<param name="frame_rate" type="double" value="$(arg frame_rate)" />
		<param name="pixel_clock" type="int" value="$(arg pixel_clock)" />
		<param name="flip_lr" type="bool" value="$(arg flip_lr)" />
	</node>

	<!-- =================== EXPOSURE FRONT CAMERA =================== -->
	<!-- <node pkg="zeabus_vision" name="auto_exposure_front" type="auto_exposure_front.py" output="screen">
		<param name="topic_right" type="str" value="/stereo/right/image_raw/compressed"/>
		<param name="client_right" type="str" value="ueye_cam_nodelet_$(arg nodelet_manager_name_right)/"/>
	</node> -->
</launch>